src/stepDefinitions/filesContext.js

```diff
@@ -163,110 +169,141 @@ const assertContentOfLocalFileIsNot = function(fullPathOfLocalFile, actualConten
-})
+Given(
+  'user {string} has locked file/folder {string} setting the following properties',
+  function (userId, fileName, table) {
+    const properties = table.rowsHash()
+    return webdavHelper.lockResource(userId, fileName, properties)
+  }
+)
 
 // here regex is used so that it can capture steps that use escape characters on file name
 // like : "file with \"nested\" quote"
-Given(/^user "([^"]*)" has created file "(.+)"$/, function(userId, fileName) {
+Given(/^user "([^"]*)" has created file "(.+)"$/, function (userId, fileName) {
   // the captured regex won't filter the backslash present in steps with folder name "file with \"nested\" quote"
   // so we need to replace it and send it as `file with "nested" quote`
-  fileName = fileName.replaceAll("\\", "")
+  fileName = fileName.replaceAll('\\', '')
   return webdavHelper.createFile(userId, fileName, '')
 })
 
 // here regex is used so that it can capture steps that use escape characters on folder name
 // like : "folder with \"nested\" quote"
-Given(/^user "([^"]*)" has created folder "(.+)"$/, function(userId, folderName) {
+Given(/^user "([^"]*)" has created folder "(.+)"$/, function (userId, folderName) {
   // the captured regex won't filter the backslash present in steps with folder name "folder with \"nested\" quote"
   // so we need to replace it and send it as `folder with "nested" quote`
-  folderName = folderName.replaceAll("\\", "")
+  folderName = folderName.replaceAll('\\', '')
   return webdavHelper.createFolder(userId, folderName)
 })
+
+Then(
+  'as user {string} folder {string} should contain {string} files',
+  async function (userName, folderName, fileCount) {
+    const davPath = webdavHelper.createDavPath(userName, folderName)
+    const resourcesInsideFolder = await webdavHelper.propfind(davPath, userName, [])
+    let jsonResponse = convert.xml2json(resourcesInsideFolder)
+    jsonResponse = JSON.parse(jsonResponse)
+    let actualCount = jsonResponse.elements[0].elements.length
+    actualCount = actualCount - 1
+    fileCount = parseInt(fileCount)
+    return assert.strictEqual(
+      actualCount,
+      fileCount,
+      `Test Failed...Expected ${fileCount} files but got ${actualCount}`
+    )
+  }
+)
+Then(
+  'as user {string} folder {string} should contain {string} subfolders',
+  async function (userName, folderName, folder_count) {
+    const davPath = webdavHelper.createDavPath(userName, folderName)
+    const resourcesInsideFolder = await webdavHelper.propfind(davPath, userName, [])
+    let jsonResponse = convert.xml2json(resourcesInsideFolder)
+    jsonResponse = JSON.parse(jsonResponse)
+    let actualCount = jsonResponse.elements[0].elements.length
+    actualCount = actualCount - 1
+    folder_count = parseInt(folder_count)
+    return assert.strictEqual(
+      actualCount,
+      folder_count,
+      `Test Failed...Expected ${folder_count} folders but got ${actualCount}`
+    )
+  }
+)
```

Comment:
hmm...implementations are ditto.
you can avoid duplications like:
```suggestion
Then(
  'as user {string} folder {string} should contain {string} items',
  async function (userName, folderName, itemsCount) {
    const davPath = webdavHelper.createDavPath(userName, folderName)
    const resourcesInsideFolder = await webdavHelper.propfind(davPath, userName, [])
    let jsonResponse = convert.xml2json(resourcesInsideFolder)
    jsonResponse = JSON.parse(jsonResponse)
    let actualCount = jsonResponse.elements[0].elements.length
    actualCount = actualCount - 1
    return assert.strictEqual(
      actualCount,
      parseInt(itemsCount),
      `Expected ${folder_count} items inside ${folderName} but found ${actualCount}`
    )
  }
)
```